google/resource_compute_ssl_policy.go

```diff
@@ -0,0 +1,281 @@
+			// CUSTOM profile demands non-zero custom_features, and other profiles (i.e., not CUSTOM) demand zero custom_features
+			if diff.HasChange("profile") || diff.HasChange("custom_features") {
+				if profile.(string) == "CUSTOM" {
+					if customFeaturesCount.(int) == 0 {
+						return fmt.Errorf("Error in SSL Policy %s: the profile is set to %s but no custom_features are set.", diff.Get("name"), profile.(string))
+					}
+				} else {
+					if customFeaturesCount != 0 {
+						return fmt.Errorf("Error in SSL Policy %s: the profile is set to %s but using custom_features requires the profile to be CUSTOM.", diff.Get("name"), profile.(string))
+					}
+				}
+				return nil
+			}
+			return nil
+		},
+	}
+}
+
+func resourceComputeSslPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	sslPolicy := &computeBeta.SslPolicy{
+		Name:           d.Get("name").(string),
+		Description:    d.Get("description").(string),
+		Profile:        d.Get("profile").(string),
+		MinTlsVersion:  d.Get("min_tls_version").(string),
+		CustomFeatures: convCustomFeaturesListToSlice(d.Get("custom_features").([]interface{})),
+	}
+
+	op, err := config.clientComputeBeta.SslPolicies.Insert(project, sslPolicy).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating SSL Policy: %s", err)
+	}
+
+	err = computeSharedOperationWaitTime(config.clientCompute, op, project, int(d.Timeout(schema.TimeoutDelete).Minutes()), "Creating SSL Policy")
+	if err != nil {
+		return err
+	}
+
+	d.SetId(sslPolicy.Name)
+
+	return resourceComputeSslPolicyRead(d, meta)
+}
+
+func resourceComputeSslPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+
+	// handle resource inputs when only the resource id (name) is given
+	if name == "" {
+		name = d.Id()
+	}
+
+	sslPolicy, err := config.clientComputeBeta.SslPolicies.Get(project, name).Do()
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("SSL Policy %q", name))
+	}
+
+	d.Set("name", sslPolicy.Name)
+	d.Set("description", sslPolicy.Description)
+	d.Set("min_tls_version", sslPolicy.MinTlsVersion)
+	d.Set("profile", sslPolicy.Profile)
+	d.Set("fingerprint", sslPolicy.Fingerprint)
+	d.Set("project", project)
+	d.Set("self_link", ConvertSelfLinkToV1(sslPolicy.SelfLink))
+
+	if sslPolicy.CustomFeatures != nil {
+		d.Set("custom_features", convertStringArrToInterface(sslPolicy.CustomFeatures))
+	}
+
+	d.SetId(sslPolicy.Name)
+
+	return nil
+}
+
+func resourceComputeSslPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+
+	d.Partial(true)
+
+	if d.HasChange("min_tls_version") {
```

Comment:
To answer your other question, yeah I think it makes sense to just do one update call for this instead of breaking it down. It's a bit weird since this resource seems to only support PATCH and not UPDATE, but there should also be no harm in calling PATCH with unchanged fields. `resource_compute_backend_service.go` is an example of a resource with a full update call